### PR TITLE
task vote view

### PR DESCRIPTION
**Pull request by terryli0095**
Screen Shot 2022-12-05 at 9 53 05 PM
## :pushpin: References

- **Wonder issue:**
- **Related pull-requests:**

## :cake: Checklist:

- [ ] I self-reviewed my changes
- [ ] My changes follow the style guide: https://creative-earth-33e.notion.site/TT-Wonder-Style-guide-4702a45133374148953bfcaf584120b7
- [ ] I tested my changes in Chrome

### CODE REVIEW

File: wondrous-app/components/Common/Votes/index.tsx [hunk 1]
```diff
@@ -1,94 +1,149 @@
...
-  taskId: string;
-  totalVotes: number;
   proposalStatus: string;
+  userInOrg: boolean;
+}
+function VoteOptionsRow({ votes, option, handleVote, handleUndoVote, user, voteType }) {
+  const percentage = computePercentage(votes, option);
+  const userVotedFor = votes?.userVote === option;
+  let optionDisplayName = option;
+  if (voteType === 'binary') {
```
Comment (terryli0095) [inline wondrous-app/components/Common/Votes/index.tsx:50]: actually i'll probably just change the data to yes and no?

File: wondrous-app/components/Common/Votes/index.tsx [hunk 2]
```diff
@@ -1,94 +1,149 @@
...
+    optionDisplayName = option === 'approve' ? 'Yes' : 'No';
+  }
+  return (
+    <VoteRowWrapper key={`${option}`}>
+      <VoteRowContentWrapper onClick={() => handleVote(option)}>
+        <div style={{ display: 'flex', gap: 5 }}>
```
Comment (terryli0095) [inline wondrous-app/components/Common/Votes/index.tsx:56]: should this be its own component?

File: wondrous-app/components/Common/Votes/index.tsx (hunk 1 above)
Comment (andywong418) [reply wondrous-app/components/Common/Votes/index.tsx:50]: Probably okay as is

File: wondrous-app/components/Common/Votes/index.tsx (hunk 2 above)
Comment (andywong418) [reply wondrous-app/components/Common/Votes/index.tsx:56]: You can make this to a styled component

File: wondrous-app/components/Common/Votes/index.tsx [hunk 3]
```diff
@@ -1,94 +1,149 @@
...
 }
 
-export default function VoteResults({ votes, fullScreen, taskId, totalVotes, proposalStatus }: Props) {
+export default function VoteResults({ userInOrg, proposal, fullScreen, proposalStatus }: Props) {
+  const { votes } = proposal;
+  const { totalVotes } = votes;
+  const proposalId = proposal.id;
   const user = useMe();
+
+  const canVote = user && userInOrg && proposalStatus === STATUS_OPEN; // TODO add logic for private pod proposals?
```
Comment (andywong418) [inline wondrous-app/components/Common/Votes/index.tsx:94]: It shouldn't be visible in private pod right?

File: wondrous-app/utils/helpers.ts [hunk 4]
```diff
@@ -64,6 +64,7 @@ export const getMentionArray = (content) => {
 };
 
 export const parseUserPermissionContext = (props) => {
+  // TODO add return type here
```
Comment (andywong418) [inline wondrous-app/utils/helpers.ts:67]: What kind of return type?

**Review by andywong418 — COMMENTED** [review]
Looks good - just some comments

File: wondrous-app/components/Common/Votes/index.tsx (hunk 2 above)
Comment (terryli0095) [reply wondrous-app/components/Common/Votes/index.tsx:56]: Personally I actually thing this is more clear on what it's doing

File: wondrous-app/utils/helpers.ts (hunk 4 above)
Comment (terryli0095) [reply wondrous-app/utils/helpers.ts:67]: Just should add typing on what the return is

File: wondrous-app/components/Common/Votes/index.tsx (hunk 3 above)
Comment (terryli0095) [reply wondrous-app/components/Common/Votes/index.tsx:94]: Yeah logic a bit complicated I was gonna add it later

**Pull request merged**